demos/Demo ICR.cpp

```diff
@@ -1,9 +1,504 @@
 #include "DGM.h"
 #include "DNN.h"
 
+#include <iostream>
+#include <fstream>
+#include <string>
+//using namespace std;
 namespace dgm = DirectGraphicalModels;
 
+
+float applySigmoidFunction(float val){
+    float sigmoid = 1 / (1 + exp(-val));
+    float value = (int)(sigmoid * 10000 + .5);
+    float result = (float)value / 10000;
+    return result;
+}
+
+
+
 int main() {
-	dgm::dnn::CNeuron neuron;
+//	dgm::dnn::CNeuron neuron;
+        
+    // ==================== Create Neurons ====================
+
+        const byte nStates = 10;
+        
+        const int digits = 10;
+        const int inputLayer = 784;
+        const int hiddenLayer = 60;
+        const int outputLayer = nStates; //10
+        
+        //[-0.1 to 0.1] 88%
+        float rangeMin = -0.5;
+        float rangeMax = 0.5;
+        
+        dgm::dnn::CNeuron myNeuron[inputLayer];
```

Comment:
When the default constructor will be removed, this code will produce error.
You can use `std::vector<std::shared_ptr<dgm::dnn::CNeuron>>` to declare a layer of neurons. Then the neurons may be added in a loop as

```
std::vector<std::shared_ptr<dgm::dnn::CNeuron>> vpOutputLayer;
for (int i = 0; i < nOutputNeurons; i++) vpOutputLayer.push_back(std::make_shared<dgm::dnn::CNeuron>(...));
```

Reply:
for this it is very difficult for me as i dont quite understand this very well the vector thing, however with matrices i fixed it to work after the default constructor. It would really be stressful to try and change the whole code structure to Vectors for me right now as it would take a long time and i dont have that much.
I would suggest that i continue with the next steps first then maybe by the end if have time i could make these conversions. :)

Reply:
Using vectors is very simple. I will be glad to explain the details. 
The main issue is to remove the default constructor in the neuron class.